x/ccv/provider/keeper/key_assignment_test.go

```diff
@@ -0,0 +1,494 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	cryptotestutil "github.com/cosmos/interchain-security/testutil/crypto"
+	testkeeper "github.com/cosmos/interchain-security/testutil/keeper"
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmprotocrypto "github.com/tendermint/tendermint/proto/tendermint/crypto"
+)
+
+type testAssignment struct {
+	chainID        string
+	providerAddr   sdk.ConsAddress
+	consumerAddr   sdk.ConsAddress
+	consumerPubKey tmprotocrypto.PublicKey
+	valsetUpdate   abci.ValidatorUpdate
+	vscID          uint64
+}
+
+func TestValidatorConsumerPubKeyCRUD(t *testing.T) {
+	chainID := "consumer"
+	providerAddr := sdk.ConsAddress([]byte("providerAddr"))
```

Comment:
You should use https://github.com/cosmos/interchain-security/blob/main/testutil/crypto/crypto.go to get test data

Reply:
E.g. https://github.com/cosmos/interchain-security/blob/095a0760566dc795b4b840deb5cbe189891d10c0/testutil/crypto/crypto.go#L94-L96

Reply:
In some places it was used, in others not.

In some places, `sdk.ConsAddress` was used so to make the tests easier to follow (mostly for brevity).

It was
`cryptotestutil.NewCryptoIdentityFromBytesSeed([]byte("validator-1")).SDKConsAddress()`

vs just

`sdk.ConsAddress([]byte("validator-1"))`

Reply:
@danwt What's the benefit from using crypto.go here?

Reply:
I just find it weird 

```
providerAddr:   sdk.ConsAddress([]byte("string")),
consumerPubKey: cryptotestutil.NewCryptoIdentityFromIntSeed(y).TMProtoCryptoPublicKey(),
```

instead of e.g.

```
providerAddr: cryptotestutil.NewCryptoIdentityFromIntSeed(x).SDKConsAddress(),
consumerPubKey: cryptotestutil.NewCryptoIdentityFromIntSeed(y).TMProtoCryptoPublicKey(),
```

But it's my opinion and not a big deal